flink/src/main/java/org/apache/iceberg/flink/FlinkFilters.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+    String pattern = tuple2.f1.toString();
+    Matcher matcher = STARTS_WITH_PATTERN.matcher(pattern);
+
+    // exclude special char of LIKE
+    // '_' is the wildcard of the SQL LIKE
+    if (!pattern.contains("_") && matcher.matches()) {
+      return Optional.of(Expressions.startsWith(tuple2.f0, matcher.group(1)));
+    }
+
+    return Optional.empty();
+  }
+
+  private static Optional<Expression> convertLogicExpression(BiFunction<Expression, Expression, Expression> function,
+                                                             CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
+    Optional<Expression> left = convert(args.get(0));
+    Optional<Expression> right = convert(args.get(1));
+    if (left.isPresent() && right.isPresent()) {
+      return Optional.of(function.apply(left.get(), right.get()));
+    }
+
+    return Optional.empty();
+  }
+
+  private static Optional<Expression> convertComparisonExpression(
+      BiFunction<String, Object, Expression> function, BiFunction<String, Object, Expression> reversedFunction,
+      CallExpression call) {
+    Tuple2<String, Object> tuple2 = convertBinaryExpress(call);
+    if (tuple2 != null) {
+      if (literalOnRight(call.getResolvedChildren())) {
+        return Optional.of(function.apply(tuple2.f0, tuple2.f1));
+      } else {
+        return Optional.of(reversedFunction.apply(tuple2.f0, tuple2.f1));
+      }
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  private static Optional<String> toReference(org.apache.flink.table.expressions.Expression expression) {
+    return expression instanceof FieldReferenceExpression ?
+        Optional.of(((FieldReferenceExpression) expression).getName()) :
+        Optional.empty();
+  }
+
+  private static Optional<Object> toLiteral(org.apache.flink.table.expressions.Expression expression) {
+    // Not support null literal
+    return expression instanceof ValueLiteralExpression ?
+        convertLiteral((ValueLiteralExpression) expression) :
+        Optional.empty();
+  }
+
+  private static Optional<Expression> handleNaN(BiFunction<String, Object, Expression> function,
+                                                Function<String, Expression> functionNaN,
+                                                CallExpression call) {
+    Tuple2<String, Object> tuple2 = convertBinaryExpress(call);
+    if (tuple2 == null) {
+      return Optional.empty();
+    }
+
+    String name = tuple2.f0;
+    Object value = tuple2.f1;
+
+    if (NaNUtil.isNaN(value)) {
+      return Optional.of(functionNaN.apply(name));
+    } else {
+      return Optional.of(function.apply(name, value));
+    }
+  }
+
+  private static Optional<Object> convertLiteral(ValueLiteralExpression expression) {
+    Optional<?> value = expression.getValueAs(expression.getOutputDataType().getLogicalType().getDefaultConversion());
+    return value.map(o -> {
+      if (o instanceof LocalDateTime) {
```

Comment:
For the flink's `Binary` data type,  its default java type is  `byte[]`,  while the iceberg's `BinaryLiteral`  will use `ByteBuffer`.    So we will need to convert it to ByteBuffer ?

Reply:
Pls see the iceberg expression literal types here : https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/types/Type.java#L30